e3db-crypto-android/src/main/java/com/tozny/e3db/AndroidCrypto.java

```diff
@@ -212,7 +212,7 @@ public File encryptFile(File file, byte[] secretKey) throws IOException, E3DBEnc
 
       for(int headAmt = source.read(head), nextAmt = source.read(next);
           headAmt != -1;
-          headAmt = nextAmt, head = next, nextAmt = source.read(next)) {
+          headAmt = nextAmt, head = next.clone(), nextAmt = source.read(next)) {
```

Comment:
This is kind of nuts -- and also why I actually really really like immutability as a concept.

Reply:
yup, the random byte pattern in java is odd